File: db4oj.tests/src/com/db4o/db4ounit/common/soda/InterfaceFieldConstraintTestCase.java
package com.db4o.db4ounit.common.soda;

import com.db4o.query.*;

import db4ounit.*;
import db4ounit.extensions.*;

public class InterfaceFieldConstraintTestCase extends AbstractDb4oTestCase {

	private static final int ID = 42;

	public static interface IFoo {
	}

	public static class Foo implements IFoo {
		public int _id;
		
		public Foo(int id) {
			_id = id;
		}
	}
	
	@Override
	protected void store() throws Exception {
		store(new Foo(ID));
	}

	public void testInterfaceFieldQuery() {
		Query query = newQuery(IFoo.class);
		query.descend("_id").constrain(ID);
		Assert.areEqual(1, query.execute().size());
	}
}

File: db4o.net/Db4objects.Db4o.Tests/Db4objects.Db4o.Tests/Common/Soda/InterfaceFieldConstraintTestCase.cs
/* Copyright (C) 2004 - 2011  Versant Inc.  http://www.db4o.com */

using Db4oUnit;
using Db4oUnit.Extensions;
using Db4objects.Db4o.Query;
using Db4objects.Db4o.Tests.Common.Soda;

namespace Db4objects.Db4o.Tests.Common.Soda
{
	public class InterfaceFieldConstraintTestCase : AbstractDb4oTestCase
	{
		private const int Id = 42;

		public interface IIFoo
		{
		}

		public class Foo : InterfaceFieldConstraintTestCase.IIFoo
		{
			public int _id;

			public Foo(int id)
			{
				_id = id;
			}
		}

		/// <exception cref="System.Exception"></exception>
		protected override void Store()
		{
			Store(new InterfaceFieldConstraintTestCase.Foo(Id));
		}

		public virtual void TestInterfaceFieldQuery()
		{
			IQuery query = NewQuery(typeof(InterfaceFieldConstraintTestCase.IIFoo));
			query.Descend("_id").Constrain(Id);
			Assert.AreEqual(1, query.Execute().Count);
		}
	}
}
